tedana/metrics/external_regressor_fits.py

```diff
@@ -0,0 +1,751 @@
+        regressors except the motion regressors. This is for the F test which compares
+        the variance explained with the full model to the variance explained if the
+        regressors-of-interest for the partial model are removed.
+    """
+    # The category titles to group each regressor
+    if "f_stats_partial_models" in external_regressor_dict:
+        partial_models = external_regressor_dict["f_stats_partial_models"]
+    else:
+        partial_models = []
+
+    # All regressor labels from the data frame
+    regressor_labels = external_regressors.columns
+
+    detrend_regressors_arr = detrend_regressors.to_numpy()
+    regressor_models = {
+        "base": detrend_regressors_arr,
+        "full": np.concatenate(
+            (detrend_regressors_arr, stats.zscore(external_regressors.to_numpy(), axis=0)), axis=1
+        ),
+    }
+
+    for pmodel in partial_models:
+        # For F statistics, the other models to test are those that include everything EXCEPT
+        # the category of interest
+        # That is "no motion" should contain the full model excluding motion regressors
+        keep_labels = set(regressor_labels) - set(external_regressor_dict[pmodel])
+        no_pmodel = f"no {pmodel}"
+        regressor_models[no_pmodel] = detrend_regressors_arr
+        for keep_label in keep_labels:
+            regressor_models[no_pmodel] = np.concatenate(
+                (
+                    regressor_models[no_pmodel],
+                    np.atleast_2d(
+                        stats.zscore(external_regressors[keep_label].to_numpy(), axis=0)
+                    ).T,
+                ),
+                axis=1,
+            )
+        keep_labels.update(set(detrend_regressors.columns))
+        LGR.info(
+            f"Size for External Regressor Partial Model '{no_pmodel}': "
+            f"{regressor_models[no_pmodel].shape}"
+        )
+        LGR.info(
+            "Regressors In Partial Model (everything but regressors of interest) "
+            f"'{no_pmodel}': {keep_labels}"
+        )
+
+    LGR.info(f"Size for full Regressor Model: {regressor_models['full'].shape}")
+    LGR.info(f"Size for base Regressor Model: {regressor_models['base'].shape}")
+
+    # vestigial codethat was used to check outputs and might be worth reviving
+    # if show_plot:
+    #     fig = plt.figure(figsize=(10, 10))
+    #     ax = fig.add_subplot(3, 2, 1)
+    #     ax.plot(detrend_regressors)
+    #     plt.title("detrend")
+    #     for idx, reg_cat in enumerate(regress_categories):
+    #         if idx < 5:
+    #             ax = fig.add_subplot(3, 2, idx + 2)
+    #             ax.plot(stats.zscore(categorized_regressors[reg_cat].to_numpy(), axis=0))
+    #             plt.title(reg_cat)
+    #     plt.savefig(
+    #         f"{prefix}_ModelRegressors.jpeg", pil_kwargs={"quality": 20}, dpi="figure"
+    #     )  # could also be saves as .eps
+
+    return regressor_models
+
+
+def fit_model(x, y, output_residual=False):
```

Comment:
This should probably go in stats since we have very similar functions there.

Reply:
done.